webextension/scripts/background.js

```diff
@@ -256,24 +256,32 @@ function getTvNews(Url, onSuccess, onFail) {
 }
 
 function getCachedWikipediaBooks(url, onSuccess, onFail) {
-  let cacheWikipediaBooks = wikipediaBooksCache[url]
+  let cacheWikipediaBooks = wikipediaBooksCache.get(url)
   if (cacheWikipediaBooks) {
     onSuccess(cacheWikipediaBooks)
   } else {
     getWikipediaBooks(url, (json) => {
-      wikipediaBooksCache[url] = json
+      if (wikipediaBooksCache.size > 2) {
+        let first_key = wikipediaBooksCache.entries().next().value[0]
```

Comment:
This may also work: `let first_key = wikipediaBooksCache.keys().next().value`

Reply:
Thank you. I will make the change.